File: ios/stereoCamera/stereoCamera/Comm/Commands/Ping.swift
//
//  Ping.swift
//  stereoCamera
//
//  Created by hallmarklabs on 6/15/18.
//  Copyright © 2018 cody. All rights reserved.
//

import Foundation

class Ping : Command
{
    private let start:Date = Date.init()
    
    private var _value:Double = -1
    public var value:Double { get {return _value }}
    
    override init()
    {
        super.init()
        cmdtype = CommandTypes.PING
        expectsResponse = true
    }
    
    override func onResponse(command: Command)
    {
        let end:Date = Date.init();
        _value = end.timeIntervalSince(start)
        
        print ("ping took " + String(_value))
    }
}

File: android/app/src/main/java/com/munger/stereocamera/ip/command/commands/Ping.java
package com.munger.stereocamera.ip.command.commands;

import android.util.Log;

import com.munger.stereocamera.ip.command.Command;

public class Ping extends Command
{
	private long start;
	public long value;

	public Ping()
	{
		super();

		cmdtype = Type.PING;
		expectsResponse = true;

		start = System.currentTimeMillis();
		value = -1;
	}

	@Override
	public void onResponse(Command command)
	{
		long end = System.currentTimeMillis();
		value = end - start;

		Log.d("Command", "ping took " + value + " millis");
	}
}
